### PR TITLE
Channellist secondarytext

### PR DESCRIPTION
**Pull request by thenakedthunder**

### CODE REVIEW

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        () => {
+            const creationDate = moment(`2017-12-22`)
+            const result = SecondaryTextHelper.getFormattedDate(creationDate)
+            expect(result).toEqual("Friday, Dec 22nd 2017")
+        }
+    )
+})
+
+describe("getTimeOfCreation", () => {
+    it("should return the creation time with the 'zero format' when it was at midnight",
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: better description, "zero format" not good enough

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            const result = SecondaryTextHelper.getDayOfCreation(creationDate)
+            const expectedResult = creationDate.format("dddd")
+
+            expect(result).toEqual(expectedResult)
+            expect(dayNamesArray).toContain(result)
+        }
+    )
+
+    // WARNING: in the first week of the year this won't test every edge case
+    it("should return the current format without year if the creation date was at least a week before the current date",
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: only on branch of the conditional described

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+    it("should return 'Yesterday' if the creation date was a day before the current date",
+        () => {
+            const creationDate = moment().subtract(1, 'days').startOf('day')
+            const result = SecondaryTextHelper.getDayOfCreation(creationDate)
+
+            expect(result).toEqual("yesterday")
+        }
+    )
+
+    it("should return the current format without year if the creation date was within a week of the current date",
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: "current": was "correct" meant?

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+            if (creationDate.year() === moment().year())
+                expectedResult = creationDate.format("dddd, MMM Do")
+            else
+                expectedResult = creationDate.format("dddd, MMM Do YYYY")
+
+            expect(result).toEqual(expectedResult)
+        }
+    )
+
+    it("should return the current format without year if the creation date was a year before the current date",
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: Copy Paste Pattern!!! :)

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+                SecondaryTextHelper.displayCreationTime(creationMoment)
+
+            expect(result).toEqual("today at 0:02")
+        }
+    )
+
+    it("should return the correct formatted creation time",
+        () => {
+            const creationMoment =
+                moment().hour(9).minute(35).subtract(1, 'days')
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: order of calls in chain good?

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+                SecondaryTextHelper.displayCreationTime(creationMoment)
+
+            expect(result).toEqual("yesterday at 9:35")
+        }
+    )
+
+    it("should return the correct formatted creation time",
+        () => {
+            const creationMoment =
+                moment().hour(14).minute(30).subtract(6, 'days')
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: order of calls in chain good?

File: flack/__tests__/SecondaryTextHelper.spec.js [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+            const result =
+                SecondaryTextHelper.displayCreationTime(creationMoment)
+            const [dayPartOfResult, timePartOfResult] = result.split(/ (.+)/)
+
+            expect(dayNamesArray).toContain(dayPartOfResult)
+            expect(timePartOfResult).toEqual("at 14:30")
+        }
+    )
+
+    it("should return the current format without year if the creation date was at least a week before the current date",
```
Comment (thenakedthunder) [inline flack/__tests__/SecondaryTextHelper.spec.js]: rewrite description

File: flack/cypress/integration/channelListPanelTest_spec.js [hunk 8]
```diff
@@ -41,4 +46,16 @@ describe('New channel not added when dialog cancelled', function () {
...
+//    it('tests that a channel with no messages displays its creation date and creator name in the secondary text part',
+//        function () {
+
+//            cy.get('#add-new-channel-btn').click()
+//            cy.get(CHANNEL_NAME_INPUT_ID).type('mamlaszok')
+//            cy.get(SUBMIT_CHANNEL_NAME_BUTTON).click()
+
+
+//        })
+//})
```
Comment (thenakedthunder) [inline flack/cypress/integration/channelListPanelTest_spec.js]: Delete!

File: flack/src/ChannelListPanel/SecondaryTextHelper.tsx [hunk 9]
```diff
@@ -1,66 +1,91 @@
-import { Channel } from "../Channel";
-import { Data } from "popper.js";
+import { Channel } from "../Channel"
+import moment, { Moment } from "moment"
+
+// TODO: CHECK FOR "PURENESS" AND MUTABILITY!!!
```
Comment (thenakedthunder) [inline flack/src/ChannelListPanel/SecondaryTextHelper.tsx]: Damn right
